server.js

```diff
@@ -0,0 +1,14 @@
+import { createServer } from "http";
+import { render } from "./dist/App.js";
+
+createServer((req, res) => {
+  const { html } = render({ url: req.url });
+
+  res.write(`
+    <!DOCTYPE html>
+    <div id="app">${html}</div>
+    <script src="/dist/bundle.js"></script>
+  `);
+
+  res.end();
+}).listen(3000);
```

Comment:
Why is this file necessary? As we're using static-hosting, we do _noy_ have a server that we can deploy like this. Was this file added in mistake when initially testing things?

Reply:
Without this file the reload function will not work on different route other than the main route. 
You can test this by: 
1. Delete this file and hydrate: true in both `main.ts` and `rollup.config `
2. Spin up the application 
3. Go to localhost:5000/testing 
4. Refresh the page and the error will show

Reply:
I followed these steps but my local copy still works. I've spun up the application using `yarn run dev` which delegates to `sirv -s`. Either way, `serve.js` would not work in Vercel, where we're deploying the application

Reply:
here is the doc from the author, [](https://github.com/EmilTholin/svelte-routing )

Reply:
I see, and that usage scenario is _not_ applicable in this project. The main usage docs assume server-side rendering, but we don't (and can't) have that configured. Hence, this file remains unused.

Reply:
Yes, you are correct and same for the comment bellow hydrate: true in both main.ts and rollup.config. But I am not sure why it wasn't working without without these config last night. But yeah since they are redundant, I will take them out.